### ISSUE TITLE
Functional test feature_fedpeg.py failing

### ISSUE THREAD

**Issue by sekulicd** [ElementsProject/elements#998]
After building project(master branch) i am trying to run functional test feature_fedpeg.py, but I am getting bellow error:

````
/usr/local/bin/python3.7 /Users/{USER}/CLionProjects/tmp/elements/test/functional/feature_fedpeg.py
2021-05-11T09:34:06.285000Z TestFramework (INFO): Initializing test directory /var/folders/q2/t3b78q_d1d9br0p8q1zqd8nc0000gn/T/bitcoin_func_test_mdka9ou5
Node 0 started
Node 1 started
Node 2 started
Node 3 started
Setting up network done
2021-05-11T09:34:13.037000Z TestFramework (INFO): sidechain info: {'fedpegscript': '512103dff4923d778550cc13ce0d887d737553b4b58f4e8e886507fc39f5e447b2186451ae', 'current_fedpeg_programs': ['a9142b5db5e644c209f5b30c82c5e887f8702bc3f1f787', 'a9142b5db5e644c209f5b30c82c5e887f8702bc3f1f787'], 'current_fedpegscripts': ['512103dff4923d778550cc13ce0d887d737553b4b58f4e8e886507fc39f5e447b2186451ae', '512103dff4923d778550cc13ce0d887d737553b4b58f4e8e886507fc39f5e447b2186451ae'], 'pegged_asset': '7e5948653ae2a6d3217bd267fc6f07cb32554c92f8048a9febebe4787ec6738c', 'min_peg_diff': '7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff', 'parent_blockhash': 'cd179c84c35f51825f20a3b91a18d45f0c53b5ceb744a5b6ef8f0babe809396f', 'parent_chain_has_pow': False, 'enforce_pak': False, 'pegin_confirmation_depth': 10, 'parent_chain_signblockscript_asm': '1', 'parent_chain_signblockscript_hex': '51', 'parent_pegged_asset': 'b2e15d0d7a0c94e4e2ce0fe6e8691b9e451377f6e46e8045a86f7c4b5d4f0f23'}
2021-05-11T09:34:13.235000Z TestFramework (ERROR): Unexpected exception caught during testing
Traceback (most recent call last):
  File "/Users/{USER}/CLionProjects/tmp/elements/test/functional/test_framework/authproxy.py", line 108, in _request
    return self._get_response()
  File "/Users/{USER}/CLionProjects/tmp/elements/test/functional/test_framework/authproxy.py", line 168, in _get_response
    http_response = self.__conn.getresponse()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/http/client.py", line 1336, in getresponse
    response.begin()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/http/client.py", line 306, in begin
    version, status, reason = self._read_status()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/http/client.py", line 275, in _read_status
    raise RemoteDisconnected("Remote end closed connection without"
http.client.RemoteDisconnected: Remote end closed connection without response
 
During handling of the above exception, another exception occurred:
 
Traceback (most recent call last):
  File "/Users/{USER}/CLionProjects/tmp/elements/test/functional/test_framework/test_framework.py", line 126, in main
    self.run_test()
  File "/Users/{USER}/CLionProjects/tmp/elements/test/functional/feature_fedpeg.py", line 223, in run_test
    txid1 = parent.sendtoaddress(addr, 24)
  File "/Users/{USER}/CLionProjects/tmp/elements/test/functional/test_framework/coverage.py", line 47, in __call__
    return_val = self.auth_service_proxy_instance.__call__(*args, **kwargs)
  File "/Users/{USER}/CLionProjects/tmp/elements/test/functional/test_framework/authproxy.py", line 144, in __call__
    response, status = self._request('POST', self.__url.path, postdata.encode('utf-8'))
  File "/Users/{USER}/CLionProjects/tmp/elements/test/functional/test_framework/authproxy.py", line 113, in _request
    self.__conn.request(method, path, postdata, headers)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/http/client.py", line 1244, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/http/client.py", line 1290, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/http/client.py", line 1239, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/http/client.py", line 1026, in _send_output
    self.send(msg)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/http/client.py", line 966, in send
    self.connect()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/http/client.py", line 938, in connect
    (self.host,self.port), self.timeout, self.source_address)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/socket.py", line 727, in create_connection
    raise err
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/socket.py", line 716, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [Errno 61] Connection refused
2021-05-11T09:34:13.300000Z TestFramework (INFO): Stopping nodes
2021-05-11T09:34:13.301000Z TestFramework.node0 (ERROR): Unable to stop node.
Traceback (most recent call last):
  File "/Users/{USER}/CLionProjects/tmp/elements/test/functional/test_framework/test_node.py", line 335, in stop_node
    self.stop(wait=wait)
  File "/Users/{USER}/CLionProjects/tmp/elements/test/functional/test_framework/coverage.py", line 47, in __call__
    return_val = self.auth_service_proxy_instance.__call__(*args, **kwargs)
  File "/Users/{USER}/CLionProjects/tmp/elements/test/functional/test_framework/authproxy.py", line 144, in __call__
    response, status = self._request('POST', self.__url.path, postdata.encode('utf-8'))
  File "/Users/{USER}/CLionProjects/tmp/elements/test/functional/test_framework/authproxy.py", line 107, in _request
    self.__conn.request(method, path, postdata, headers)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/http/client.py", line 1244, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/http/client.py", line 1255, in _send_request
    self.putrequest(method, url, **skips)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/http/client.py", line 1108, in putrequest
    raise CannotSendRequest(self.__state)
http.client.CannotSendRequest: Request-sent
Traceback (most recent call last):
  File "/Users/{USER}/CLionProjects/tmp/elements/test/functional/feature_fedpeg.py", line 727, in <module>
    FedPegTest().main()
  File "/Users/{USER}/CLionProjects/tmp/elements/test/functional/test_framework/test_framework.py", line 149, in main
    exit_code = self.shutdown()
  File "/Users/{USER}/CLionProjects/tmp/elements/test/functional/test_framework/test_framework.py", line 278, in shutdown
    self.stop_nodes()
  File "/Users/{USER}/CLionProjects/tmp/elements/test/functional/test_framework/test_framework.py", line 537, in stop_nodes
    node.wait_until_stopped()
  File "/Users/{USER}/CLionProjects/tmp/elements/test/functional/test_framework/test_node.py", line 378, in wait_until_stopped
    wait_until_helper(self.is_node_stopped, timeout=timeout, timeout_factor=self.timeout_factor)
  File "/Users/{USER}/CLionProjects/tmp/elements/test/functional/test_framework/util.py", line 262, in wait_until_helper
    if predicate():
  File "/Users/{USER}/CLionProjects/tmp/elements/test/functional/test_framework/test_node.py", line 369, in is_node_stopped
    "Node returned non-zero exit code (%d) when stopping" % return_code)
AssertionError: [node 0] Node returned non-zero exit code (-10) when stopping
[node 3] Cleaning up leftover process
[node 1] Cleaning up leftover process
[node 0] Cleaning up leftover process
 
Process finished with exit code 1
 
````

I am using macOS Catalina, 10.15.7.

**Comment by sekulicd** [comment]
> We've seen similar issues on MacOS e.g. #782
> could you try do clean rebuild with extra configure options `CFLAGS=-fno-stack-check CXXFLAGS=-fno-stack-check` ?

yes, using additional flags resolves the issue.

**Comment by apoelstra** [comment]
Closing as this should be fixed by #1001
